src/learner/dqn.jl

```diff
@@ -55,11 +57,11 @@ function DQN(net; kargs...)
 end
 function defaultbuffer(learner::Union{DQN, DeepActorCritic}, env, preprocessor)
     state = preprocessstate(preprocessor, getstate(env)[1])
-    ArrayStateBuffer(capacity = typeof(learner) <: DQN ? learner.replaysize :
-                                                         learner.nsteps + learner.nmarkov, 
-                     arraytype = typeof(state).name.wrapper,
-                     datatype = typeof(state[1]),
-                     elemshape = size(state))
+    action = sample(actionspace(env))
+    CircularTurnBuffer{Turn{typeof(state), typeof(action), Float64, Bool}}(
+        typeof(learner) <: DQN ? learner.replaysize : learner.nsteps + learner.nmarkov,
```

Comment:
Should be ```typeof(learner) <: DQN ? learner.replaysize : learner.nsteps```. This also fixes the second test of linfuncapprox.

Reply:
resolved not